File: onboarder/src/main/java/com/chyrta/onboarder/OnboarderAdapter.java
package com.chyrta.onboarder;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;

import java.util.ArrayList;
import java.util.List;

public class OnboarderAdapter extends FragmentStatePagerAdapter {

    List<OnboarderPage> pages = new ArrayList<>();

    public OnboarderAdapter(FragmentManager fm) {
        super(fm);
    }

    public void addOnboarderPage(OnboarderPage onboarderPage) {
        pages.add(onboarderPage);
    }

    @Override
    public Fragment getItem(int position) {
        return OnboarderFragment.newInstance(pages.get(position));
    }

    @Override
    public int getCount() {
        return pages.size();
    }


}

File: onboarder/src/main/java/com/chyrta/onboarder/OnboarderAdapter.kt
package com.chyrta.onboarder

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentStatePagerAdapter

class OnboarderAdapter(fm: FragmentManager) : FragmentStatePagerAdapter(fm) {

    internal var pages: MutableList<OnboarderPage> = mutableListOf<OnboarderPage>()

    fun addOnboarderPage(onboarderPage: OnboarderPage) {
        pages.add(onboarderPage)
    }

    override fun getItem(position: Int): Fragment {
        return OnboarderFragment.newInstance(pages[position])
    }

    override fun getCount(): Int {
        return pages.size
    }

}
